### PR TITLE
Adds more safeguards against subsystems crashing, and a ton of other subsystem related changes

### PR DESCRIPTION
**Pull request by Gboster-0**
## About The Pull Request

<details>
<summary>Ports the following PR's</summary>

- https://github.com/tgstation/tgstation/pull/80304
- https://github.com/tgstation/tgstation/pull/63222
- https://github.com/tgstation/tgstation/pull/57800
- https://github.com/tgstation/tgstation/pull/59792
- https://github.com/tgstation/tgstation/pull/59813
- https://github.com/tgstation/tgstation/pull/59974
- https://github.com/tgstation/tgstation/pull/59843
- https://github.com/tgstation/tgstation/pull/56453
- https://github.com/tgstation/tgstation/pull/59915
- https://github.com/tgstation/tgstation/pull/60413
- https://github.com/tgstation/tgstation/pull/57381
- https://github.com/tgstation/tgstation/pull/57083
- https://github.com/tgstation/tgstation/pull/57211
- https://github.com/tgstation/tgstation/pull/57478
- https://github.com/tgstation/tgstation/pull/57511
- https://github.com/tgstation/tgstation/pull/60846
- https://github.com/tgstation/tgstation/pull/63118
- https://github.com/tgstation/tgstation/pull/64498

</details>

Basically a lotta lot of stuff related to subsystems, atmos and other things
Adds a subsystem for speaking

## Why It's Good For The Game

People should be more likelly to automatically reconnect successfully to the server upon a restart
Subsystems should lag less?
Also you can now delete the master subsystem and the server will automatically recover/create a new one instead of crashing,

### CODE REVIEW

**Comment by Kitsunemitsu** [comment]
Oh man this is going to be a very long review

**Review by Kitsunemitsu — APPROVED** [review]
So we had an admin literally delete the master controller and it did not crash the server so this is probably good to go.
No crashes related to it so far.
